### PR TITLE
Messaging smoke test

### PR DESCRIPTION
**Pull request by gauntface**
### Testing

This adds a set of end to end integration tests to ensure that the messaging SDK can at least send and receive expected messages.

### CODE REVIEW

**Comment by gauntface** [comment]
@jshcrowthe the test looks like a flakey test not caused by this PR (i.e. the integration tests aren't run so everything is ok).

**Comment by gauntface** [comment]
@jshcrowthe what would you like me to do with this? Is it blocked on your other testing changes?

**Comment by jshcrowthe** [comment]
@gauntface I will integrate them w/ the rest of the scripts. Expect a follup commit here in a min.

**Comment by gauntface** [comment]
@jshcrowthe I think this works. I can't run gulp test so no idea if your new stuff works. Tried gulp test:integration like the readme but I can't find it in my gulp tasks. Nothing shows up in code search either: https://github.com/firebase/firebase-js-sdk/search?l=JavaScript&q=test%3Aintegration&type=&utf8=%E2%9C%93

**Comment by gauntface** [comment]
Pinging @jshcrowthe

File: integration/messaging/runner.sh [hunk 1]
```diff
@@ -1,2 +1,46 @@
+#!/bin/bash
+
+set -ev
+
+# Variables
```
Comment (gauntface) [inline integration/messaging/runner.sh]: Why is any of these needed?

**Pull request merged**